sacro-app/main.js

```diff
@@ -31,16 +36,32 @@ function findAppPath() {
 }
 
 
-function createWindow() {
-  const win = new BrowserWindow({
-    width: 1024,
-    height: 768,
-  });
-  //win.maximize();
+async function createWindow() {
+ 
+  let freePort = null;
+  try {
+    freePort = await portfinder.getPortPromise({ port: 8001 });
+  } catch (err) {
+    console.error("Failed to obtain a port", err);
+    app.quit();
+  }
+  const serverUrl = `http://127.0.0.1:${freePort}/`;
 
   // Spawn the server process
   p = findAppPath()
-  const serverProcess = spawn(p)
+  const serverProcess = spawn(p, {env: {
+    SACRO_APP_TOKEN: RANDOM_SECRET,
+    PORT: freePort,
```

Comment:
The race free way to do this would be not supply a port, let the OS choose one and then ask which one it chose. But how easy this is depends on how easy it would be to log this port from the Python process and then read it back out in JS.

Reply:
I just stole this wholesale from Simon Willison's electron app.

However, opening a network port triggers windows firewall pop up. If you approve it, it remembers that port, and doesn't ask again. However, if it was random, it might trigger that pop up everytime.

Normally, we'd just use a unix socket and make this all be a non-issue.  But there's no easy way to do that in windows that's supported (by node js client and python servers, anyway)